optuna/samplers/_tpe/sampler.py

```diff
@@ -214,6 +214,21 @@ def objective(trial):
             If this is :obj:`True` and ``multivariate=True``, a warning message is emitted when
             the value of a parameter is sampled by using an independent sampler.
             If ``multivariate=False``, this flag has no effect.
+        constant_liar:
```

Comment:
In nii, they can configure constant_liar by following arguments:

> classArgs requirements:

> optimize_mode (maximize or minimize, optional, default = maximize) - If 'maximize', tuners will try to maximize metrics. If 'minimize', tuner will try to minimize metrics.
>parallel_optimize (bool, optional, default = False) - If True, TPE will use the Constant Liar algorithm to optimize parallel hyperparameter tuning. Otherwise, TPE will not discriminate between sequential or parallel situations.
>constant_liar_type (min or max or mean, optional, default = min) - The type of constant liar to use, will logically be determined on the basis of the values taken by y at X. There are three possible values, min{Y}, max{Y}, and mean{Y}.

https://github.com/microsoft/nni/blob/e85f029bd4e4b1bdf3e679893fb6447e4d6b2c79/docs/en_US/Tuner/HyperoptTuner.rst

Reply:
Do mean mean we should do that in Optuna as well? I tried to address that in the PR description.

Reply:
Looks reasonable. @hvy How about adding `constant_liar_type` argument?

Reply:
Do you mean introducing another parameter in addition to `constant_liar` ? One should be enough since it includes the boolean information. However, again, I'm worried that it'll be more confusing than useful for many users.

Reply:
> One should be enough since it includes the boolean information.

You are right. It's enough to include the type of constant liar information in `constant_liar`. 

> I'm worried that it'll be more confusing than useful for many users.

I see. It's OK for me to restrict the type of constant liar to `inf` in this PR. I think it will be enough if we work on it after the users request to change the type of constant liar.

Reply:
Sorry, it is just FYI.

And, especially, I noticed that they named the argument from the functionality `parallel_optimize`, not from the implementation `constant_liar`.
 
The argument `constant_liar` is a little bit difficult for users to recognize that the functionality is for distributed optimization, although they can easily understand that if they read docstrings.

In my opinion, we can use `constant_liar` here, because users will read the docstrings thoroughly if they want to use the functionality.